src/GraphQL/Server.hs

```diff
@@ -77,11 +76,21 @@ infixr 8 :<>
 data a :<|> b = a :<|> b
 infixr 8 :<|>
 
+
+-- Result collects errors and values at the same time unless a handler
+-- tells us to bail out in which case we stop the processing
+-- immediately.
+-- TODO: Make QueryError a Monoid and aggregate according to that.
+data Result = Result [QueryError] GValue.Value
+ok :: GValue.Value -> Result
+ok v = Result [] v
```

Comment:
You should probably make `Result` implement `Functor`, `Applicative`, and `Monad`. Then `ok` becomes `pure`. We can still keep around in its specific form, but it'll make a bunch of the code below simpler (I think).